pkg/network/dns_snooper.go

```diff
@@ -188,7 +172,14 @@ func (s *SocketFilterSnooper) processPacket(data []byte, ts time.Time) {
 
 func (s *SocketFilterSnooper) pollPackets() {
 	for {
-		data, captureInfo, err := s.source.ZeroCopyReadPacketData()
+		err := s.source.VisitPacket(func(data []byte, timestamp time.Time) error {
```

Comment:
I'm curious if this closure has a negative performance impact. It may be better to have `processPacket` return an `error` and just use `s.source.VisitPacket(s.processPacket)` here.

Reply:
I don't think we have a sensitive enough way to measure this, but that's a pretty easy change, so I'll make it!